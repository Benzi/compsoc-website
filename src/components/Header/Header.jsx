import * as React from "react"


// Styles
import styles from "./Header.module.scss"
import logoalt from "../../images/logo-inverted.svg"
import logo from "../../images/logo.svg"


// Load components from React-Bootstrap
import Navbar from "react-bootstrap/Navbar"
import Nav from "react-bootstrap/Nav"

const Header = (props) => {
    return (
    <header>

    <Navbar bg="light" expand="lg">
      <div className="container">
        <Navbar.Brand href="#home">
          <img
            src={logo}
            width="120"
            height="30"
            className="d-inline-block align-top"
            alt="Computing Society"
          />
        </Navbar.Brand>      <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link href="#home">Home</Nav.Link>
            <Nav.Link href="#link">Events</Nav.Link>
            <Nav.Link href="#link">About</Nav.Link>
            <Nav.Link href="#link">Our Team</Nav.Link>
            <Nav.Link href="#link">Sponsors</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </div>
    </Navbar>


      <p>This is a header</p>
    </header>

  )
}

export default Header