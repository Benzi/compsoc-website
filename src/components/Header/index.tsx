import { Link } from "gatsby"
import React from "react"
import { Icon, Navbar } from "react-materialize"
import compsoc from "./compsoc.svg"
import styles from "./index.module.scss"

const Header = () => {
  return (
    <Navbar
      alignLinks="right"
      centerChildren
      brand={
        <Link className="brand-logo" to="/">
          <img src={compsoc} className={styles.logo} />
        </Link>
      }
      menuIcon={<Icon>menu</Icon>}
    >
      <Link to="/">Home</Link>
    </Navbar>
  )
}

export { Header }
