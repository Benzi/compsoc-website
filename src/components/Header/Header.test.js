import React from "react"
import renderer from "react-test-renderer"

import Header from "./Header"


// Using SnapShot testing
// This ensures that a change we makes elesewhere does not change the behaviour of existing components
// If the test fails due to an INTENTIONAL change run `jest --updateSnapshot`
describe("Header", () => {

    // What we are testing
    it("renders correct", () => {
        const tree = renderer
    
        // Crate the component
        .create(<Header />)
        .toJSON() // Convert to JSON

        // Check if it matches what we expect by checking a snapshot from the past
        // https://jestjs.io/docs/en/snapshot-testing
        expect(tree).toMatchSnapshot()
    })

})