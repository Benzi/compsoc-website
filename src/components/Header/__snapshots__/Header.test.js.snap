// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correct 1`] = `
<div>
  <p>
    This is a header
  </p>
</div>
`;
